src/DataManagement/RecurringUpdater.cs

```diff
@@ -0,0 +1,125 @@
+                var newEntry = new Entry(entry.Title, entry.Amount, nextMonth, false, entry.Importance);
+                tempList.Add(newEntry);
+            }
+
+            /*Adding last entry, which has to keep isMonthly*/
+            nextMonth = TimeManager.MoveToNextMonth(nextMonth);
+            var newestEntry = new Entry(entry.Title, entry.Amount, nextMonth, true, entry.Importance);
+            tempList.Add(newestEntry);
+            if (!entryManager.AddRange(tempList))
+            {
+                ExceptionHandler.Log("Couldn't add list of entries in updater: " + tempList);
+                return false;
+            }
+
+            //Moved here since no point to edit in each cycle rotation
+            var editedOriginalEntry = new Entry(entry.Id, entry.UserId, entry.Title, entry.Amount, entry.Date, false, entry.Importance);
+            if (entryManager.Edit(entry, editedOriginalEntry))
+            {
+                return true;
+            }
+
+            ExceptionHandler.Log("Couldn't edit entry in updater:" + editedOriginalEntry);
+            return false;
+        }
+
+        public static bool UpdateEntry(IEntry entry, int userId, EntryType entryType)
+        {
+            var nextMonth = entry.Date;
+            var months = TimeManager.DifferenceInMonths(laterTime: DateTime.Today, earlierTime: nextMonth);
+            // one less since last one has to keep isMonthly = true;
+            var tempList = new EntryList(entryType);
+            for (var i = 0; i < months - 1; i++)
+            {
+                //Adding new entry for each month according to date difference
+                nextMonth = TimeManager.MoveToNextMonth(dateTime: nextMonth);
+                var newEntry = new Entry(entry.Title, entry.Amount, nextMonth, false, entry.Importance);
+                tempList.Add(newEntry);
+            }
+
+            /*Adding last entry, which has to keep isMonthly*/
+            nextMonth = TimeManager.MoveToNextMonth(nextMonth);
+            var newestEntry = new Entry(entry.Title, entry.Amount, nextMonth, true, entry.Importance);
+            tempList.Add(newestEntry);
+            if (!EntryDbUpdater.AddRange(tempList, userId))
+            {
+                ExceptionHandler.Log("Couldn't add list of entries in updater: " + tempList);
+                return false;
+            }
+
+            //Moved here since no point to edit in each cycle rotation
+            var editedOriginalEntry = new Entry(entry.Id, entry.UserId, entry.Title, entry.Amount, entry.Date, false, entry.Importance);
+            if (EntryDbUpdater.Edit(entry.Id, editedOriginalEntry, entryType))
+            {
+                return true;
+            }
+
+            ExceptionHandler.Log("Couldn't edit entry in updater:" + editedOriginalEntry);
+            return false;
+        }
+
+
+
+        public static bool AddMonthlyEntry(IEntry entry, int userId, EntryType entryType)
```

Comment:
Where do we use `AddMonthlyEntry` since `Update...` seems to be used for already existing entries

Reply:
Update Monthly Entry is for private use in class, it used when update a whole Entry List ( basically just the old version of how we used to do updates) and Add Monthly Entry is for standalone use, when we are adding a new entry and it is monthly if that makes sense

Reply:
So let's I'm adding Entry in front code and if it is Monthly expenses, I should call same method as for all expenses or this should be called instead?

Reply:
At the moment you'd have to manually call this method, but soon I'll integrate it so it you can just add it like normal

Reply:
Oh ok, then it is awesome